service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -240,11 +240,17 @@ private void buildNarrativeStatementForObservationComments(
         var narrativeStatementObservationComments = observations.stream()
             .filter(Observation::hasCode)
             .filter(this::hasCommentNote)
-            .map(Observation::getComment)
+            .map(observation -> {
+                final var comment = observation.getComment();
+                if(StringUtils.isNotBlank(comment)) {
+                    return buildNarrativeStatementForDiagnosticReport(
+                        issuedElement, CommentType.USER_COMMENT.getCode(), comment, confidentialityService.generateConfidentialityCode(observation).orElse(null)
+                    );
+                }
+
+                return null;
+            })
```

Comment:
Not a fan of providing the implementation here of this `@FunctionalInterface` - if we could separate this out it would be great.

Reply:
This refactoring will be done in a separate PR.